### PR TITLE
Ridley Leisy - Unit 2 - Classification 1

### PR DESCRIPTION
**Pull request by RidleyLeisy**

### CODE REVIEW

**Comment by axrd** [comment]
# Module 01 - Logistic Regression & One-Hot Encoding  

**Overall**: Well done! You're accuracy score is pretty darn good 👍 
**Feedback**: Comment your code! Comment it to force a good habit of discipline. Future you will likely not remember what you meant with every line of code in past notebooks. Keep him in mind. Also, use a validation set, and leverage your coefficients!

## Exploring/Cleaning data
- Try the `df.info()` method. Might help you as an initial step.
- Other than that, well done making your code functional 👍 
- Nice job spotting the 0s in 'Construction Year' 👍 

## Train/Val/Test Split
- You want to create a validation set! Give yourself another stopgap to boost your predictive power. 
- Remember that you want to make sure that any preprocessing that you do on the data is reflected in both the train and test sets!

## One-Hot Encoding & Scaling
- Nicely done! Kudos for evaluating cardinality of your features👍 

## Logistic Regression Model
 - Yep. That's how you do it 👍 

## Coefficients
- You need to retrieve your coefficients and plot them! 👎 
- In the Lecture notebook, Ryan H wrote a nifty way to do this:
```
# Import Matplotlib (inline)
%matplotlib inline
import matplotlib.pyplot as plt

coefficients = pd.Series(model.coef_[0], X_train_encoded.columns)
plt.figure(figsize=(10, 10))
coefficients.sort_values().plot.barh();
```

## Submit to Kaggle Competition
- Aww yeah! How's it feel to be on the leaderboard?

**Comment by axrd** [comment]
# Module 02 - Decision Trees  

**Overall**: Well done! Congratulations on breaking 80% Accuracy! 📈 
**Feedback**: **_Comment your code!_** Nicely done jumping ahead and trying out a Random 🌲 🌳 🌳 🌲 👍 

## More Exploring/Cleaning data
- Dig deeper. Lookout for missing or incorrect values in other Features. 

## Implement a Decision Tree Classifier
- Yep. Nicely done 👍 + 💯 for going ahead and implementing a forest!
- Plot the feature importances to help guide your approach
```
# Get feature importances
encoder = pipeline.named_steps['onehotencoder']
tree = pipeline.named_steps['decisiontreeclassifier']
feature_names = encoder.transform(X_val).columns
importances = pd.Series(tree.feature_importances_, feature_names)

# Plot feature importances
n = 20
plt.figure(figsize=(10,n/2))
plt.title(f'Top {n} features')
importances.sort_values()[-n:].plot.barh(color='grey');
```

## Submit New Predictions to Kaggle
- I see you climbing the leaderboard 🙌

**Comment by axrd** [comment]
# Module 03 - Random Forests, Ordinal Encoding  

**Overall**: Good work! Congratulations on holding 8th in the Public Leaderboard!
**Feedback**: Organize your forked repo a bit better. Your Module 3 assignemnt probably shouldn't be committed into the Module 1 directory. Almost missed it! Also, I'm appreciating the additional commenting put don't forget to do it inside your functions as well 🆗 ?

## More Exploring/Cleaning/Feature Engineering
- Solid progress, but there's more you can do here. 
- Kudos for reducing dimensionality as part of your iteration

## Implement a Random Forest Classifier
- Yep. Nicely done 👍 + 💯 for going ahead and trying XGBoost! 🚀 

## Try Ordinal Encoding
- Great job making this part of a function and segregating it from One-hot

## Submit New Predictions to Kaggle
- 7 hours to go! Get it 💪

**Comment by axrd** [comment]
# Module 04 - Confusion Matrix, Precision & Recall

## Final Kaggle Submission:

**Overall**: Your model had some decent predictive power, but it could be improved. I'd encourage you to re-create some of the lecture notebooks (they'll break 81%). That being said, remember that generalzing to new data (public vs private leaderboar) has a substantial element of random chance. 

More importantly I would talk with your PM during your 1:1 to make sure you absorbed the material as well as you wanted to this time around. Any gaps are better to fill now than allow them to expand!

Oh and if you want to learn how to combat that problem about generalizing to new data, look up Cross Validation for a preview of things to come 👍
